build-deps.zsh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env zsh
+
+builtin emulate -L zsh
+setopt EXTENDED_GLOB
+setopt PUSHD_SILENT
+setopt ERR_EXIT
+setopt ERR_RETURN
+setopt NO_UNSET
+setopt PIPE_FAIL
+setopt NO_AUTO_PUSHD
+setopt NO_PUSHD_IGNORE_DUPS
+setopt FUNCTION_ARGZERO
+#setopt WARN_CREATE_GLOBAL
+#setopt WARN_NESTED_VAR
+#setopt XTRACE
```

Comment:
Those are default shell options helpful for debugging and development. I kept them in there so that maintainers/contributors can easily enable them to debug any issues that might occur.

To answer the last question, those should be familiar to anyone creating scripts in Zsh.

Reply:
Got it.